static/app/views/explore/hooks/useTraceItemAttributeKeys.tsx

```diff
@@ -36,76 +20,33 @@ export function useTraceItemAttributeKeys({
   traceItemType,
   projects,
 }: UseTraceItemAttributeKeysProps) {
-  const organization = useOrganization();
-  const {selection} = usePageFilters();
-
-  const path = `/organizations/${organization.slug}/trace-items/attributes/`;
-  const endpointOptions = {
-    query: {
-      project: defined(projects)
-        ? projects.map(project => project.id)
-        : selection.projects,
-      environment: selection.environments,
-      ...normalizeDateTimeParams(selection.datetime),
-      itemType: traceItemType,
-      attributeType: type,
-    },
-  };
-
-  const result = useApiQuery<Tag[]>([path, endpointOptions], {
-    enabled,
-    staleTime: 0,
-    refetchOnWindowFocus: false,
-    retry: false,
+  const queryOptions = useTraceItemAttributeKeysQueryOptions({
+    traceItemType,
+    type,
+    projectIds: defined(projects) ? projects.map(project => project.id) : undefined,
   });
 
-  const attributes: TagCollection = useMemo(() => {
-    const allAttributes: TagCollection = {};
-
-    for (const attribute of result.data ?? []) {
-      if (isKnownAttribute(attribute)) {
-        continue;
-      }
-
-      // EAP spans contain tags with illegal characters
-      // SnQL forbids `-` but is allowed in RPC. So add it back later
-      if (
-        !/^[a-zA-Z0-9_.:-]+$/.test(attribute.key) &&
-        !/^tags\[[a-zA-Z0-9_.:-]+,number\]$/.test(attribute.key)
-      ) {
-        continue;
-      }
+  const queryKey = useMemo(
+    () => ['use-trace-item-attribute-keys', queryOptions],
+    [queryOptions]
+  );
 
-      allAttributes[attribute.key] = {
-        key: attribute.key,
-        name: attribute.name,
-        kind: type === 'number' ? FieldKind.MEASUREMENT : FieldKind.TAG,
-      };
-    }
+  const getTraceItemAttributeKeys = useGetTraceItemAttributeKeys({
+    traceItemType,
+    type,
+    projectIds: defined(projects) ? projects.map(project => project.id) : undefined,
+  });
 
-    return allAttributes;
-  }, [result.data, type]);
+  const {data, isFetching} = useQuery<TagCollection>({
```

Comment:
should we handle errors here? I assume we need to adjust `getTraceItemAttributeKeys`.